share/completions/gradle.fish

```diff
@@ -1,60 +1,276 @@
-complete -c gradle -l info -s i -d 'Set log level to info'
-complete -c gradle -l include-build -r -d 'Include specified build in composite'
-complete -c gradle -l dry-run -s m -d 'Runs build with all task actions disabled'
-complete -c gradle -l max-workers -x -d 'Configure number of concurrent workers' -a '1\t 2\t 3\t 4\t 5\t 6\t 7\t 8\t 9\t 10\t'
-complete -c gradle -l no-daemon -d 'Don\'t use deamon'
-complete -c gradle -l offline -d 'Don\'t use network resources'
-complete -c gradle -l project-prop -s P -x -d 'Set project property for build script'
-complete -c gradle -l project-dir -s p -r -d 'Specify start directory'
-complete -c gradle -l parallel -d 'Build project in parallel'
-complete -c gradle -l profile -d 'Profile build execution time'
-complete -c gradle -l project-cache-dir -r -d 'Specify project cache directory'
-complete -c gradle -l quiet -s q -d 'Only log erros'
-complete -c gradle -l recompile-scripts -d 'Force build script recompiling'
-complete -c gradle -l refresh-dependencies -d 'Refresh state of dependencies'
-complete -c gradle -l rerun-tasks -d 'Ignore previously cached dependencies'
-complete -c gradle -l full-stacktrace -s S -d 'Print out full stacktrace for all exceptions'
-complete -c gradle -l status -d 'Shows status of running andrecently stopped daemon'
-complete -c gradle -l stop -d 'Stop daemon if running'
-complete -c gradle -l continuous -s t -d 'Enable continuous build'
-complete -c gradle -l no-search-upward -s u -d 'Don\'t search in parent folders for settings file'
-complete -c gradle -l version -s v -d 'Print version'
-complete -c gradle -l exclude-task -s x -x -d 'Specify task to be excluded from execution'
-
-# https://github.com/hanny24/gradle-fish/blob/master/gradle.load
-function __cache_or_get_gradle_completion
+function __contains_gradle_build
+    test -f build.gradle -o -f build.gradle.kts
+end
+
+function __create_completion_cache_file
     # Set up cache directory
     if test -z $XDG_CACHE_HOME
         set XDG_CACHE_HOME $HOME/.cache/
     end
     mkdir -m 700 -p $XDG_CACHE_HOME/gradle-completions
 
-    set -l hashed_pwd (__fish_md5 -s $PWD)
-    set -l gradle_cache_file $XDG_CACHE_HOME/gradle-completions/$hashed_pwd
-    if not test -f $gradle_cache_file; or command test build.gradle -nt $gradle_cache_file
-        command gradle -q tasks 2>/dev/null | string match -r '^[[:alnum:]]+ - .*' | string replace ' - ' \t >$gradle_cache_file
+    string trim -- $XDG_CACHE_HOME/gradle-completions/(__fish_md5 -s $argv[1] | string split ' = ')[2]
+end
+
+##############################
+# Configure Tasks Completion #
+##############################
+
+# Outside of a Project
+function __get_gradle_default_task_completion
+    if __contains_gradle_build
+        return
     end
-    cat $gradle_cache_file
+
+    printf '%s\t%s\n' \
+        "buildEnvironment" "Displays all buildscript dependencies declared in root project." \
```

Comment:
To be honest these are still a bit verbose.

"Display buildscript dependencies"

Note the lack of the "s" - for one this is an *order* you give the command, so the imperative makes sense, for another just ditching the "s" saves a character, just like removing the "." does.

And it displaying _all_ dependencies just seems implicitly true. If it says dependencies, I would assume it's all of them, or it should tell me why it isn't all.

Reply:
Assuming you decide that it is ok to diverge from the Gradle naming convention, `all` probably could be removed also for `gradle dependencies` - that task can be reduced to display dependencies only from the specific configuration and then it is all for that configuration, but not all at all :).

Reply:
It *definitely* is okay to diverge from the exact wording of Gradle's documentation.

These things have a very different purpose.

Now "naming convention" implies that we'd e.g. use our own *terms*. That's probably not a good thing to do. What Gradle calls a "function" we should also call a "function" and not a "method" (to pick a common example).